src/robot/model/body.py

```diff
@@ -120,39 +147,40 @@ def create(self):
             f"Use item specific methods like 'create_keyword' instead."
         )
 
-    def create_keyword(self, *args, **kwargs):
-        return self._create(self.keyword_class, 'create_keyword', args, kwargs)
-
-    def _create(self, cls, name, args, kwargs):
+    def _create(self, cls: 'Type[T]', name: str, args: Iterable[Any],
+                kwargs: Mapping[str, Any]) -> T:
         if cls is None:
             raise TypeError(f"'{full_name(self)}' object does not support '{name}'.")
         return self.append(cls(*args, **kwargs))
 
-    def create_for(self, *args, **kwargs):
+    def create_keyword(self, *args, **kwargs) -> 'Keyword':
+        return self._create(self.keyword_class, 'create_keyword', args, kwargs)
+
+    def create_for(self, *args, **kwargs) -> 'For':
         return self._create(self.for_class, 'create_for', args, kwargs)
 
-    def create_if(self, *args, **kwargs):
+    def create_if(self, *args, **kwargs) -> 'If':
```

Comment:
Having `robot.model.If` as a return type is a bit problematic, because then also `robot.running` and `robot.result` side model objects use that and not their own `If`. It's nevertheless an improvement over not having a return type. I know your are working with enhancements for `create` signature and we can return to this later.